DOCSP-26099: cs is an async iterable (#498)
mongodb/docs-node: source/usage-examples/changeStream.txt

section for more information on the settings you can configure with this object.

The ``watch()`` method returns an instance of a `ChangeStream <{+api+}/classes/ChangeStream.html>`__. You can read events from
change streams by iterating over them or listening for events. 

Select the tab that corresponds to the way you want to
read events from the change stream:

.. tabs::

   .. tab::
      :tabid: Idiomatic Iteration

>       Starting in version 4.12, the ``ChangeStream`` class is an async
>       iterable. With this change, you can use ``for-await`` loops to

Reviewer: I: I don't _think_ it's accurate to say that a class is an async iterable. Based on [this page](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols#the_async_iterator_and_async_iterable_protocols), a class/object implements the async iterable protocol, and then the _object_ is the async iterable. I know this gets weird in JS, which isn't a true OOP language, but maybe we could try:

```suggestion
      Starting in version 4.12, the ``ChangeStream`` class implements the async
      iterable protocol. With this change, you can use ``for-await`` loops to
```

Or:

```suggestion
      Starting in version 4.12, ``ChangeStream`` objects are async
      iterables. With this change, you can use ``for-await`` loops to
```

   The JavaScript and TypeScript code snippets above are identical. There are no
   TypeScript specific features of the driver relevant to this use case.

> When you run this code and then make any changes to the ``haikus``

Reviewer: I: Next clause refers to a single change

S: 

```suggestion
When you run this code and then make a change to the ``haikus``
```

      const changeStream = collection.watch(pipeline, options);

Listener Function
~~~~~~~~~~~~~~~~~

The following example opens a change stream on the ``haikus`` collection in
the ``insertDB`` database. Let's create a listener function to receive and
print change events that occur on the collection.

First, open the change stream on the collection and then define a listener

> on the change stream using the ``on()`` method. Once set, generate a change

Reviewer: I: "Once set" is vague

S: "Once the [whatever] is [whatever], generate a change event..."

event by performing a change to the collection.

> To generate the change event on the collection, let's use ``insertOne()``
> method to add a new document. Since the ``insertOne()`` may run before the

Reviewer: ```suggestion
To generate the change event on the collection, let's use the ``insertOne()``
method to add a new document. Since ``insertOne()`` may run before the
```

listener function can register, we use a timer, defined as
``simulateAsyncPause`` to wait 1 second before executing the insert.

> We also use ``simulateAsyncPause`` after the insertion of the document
> to provide ample time for the listener function to receive the change
> event and for the listener to complete its execution before
> closing the ``ChangeStream`` instance using the ``close()`` method.

Reviewer: ```suggestion
We also use ``simulateAsyncPause`` after the insertion of the document.
This provides ample time for the listener function to receive the change
event and for the listener to complete its execution before
closing the ``ChangeStream`` instance using the ``close()`` method.
```

> The timers used in this example are only necessary for this demonstration
> to make sure there is enough time to register listener and have the
> listener process the event before exiting.

Reviewer: S: Put this in a Note admonition to make it stand out

```suggestion
.. note:: You don't need timers to use listener functions.

   The timers in this code example are for demonstration purposes only.
```

...or something similar.

Reviewer: I'm in favor of deleting the redundant note.
